lib/sth_database.js

```diff
@@ -1841,6 +1841,490 @@ function getNotificationInfo(data, callback) {
+                  {
+                    _id: collectionName
+                  },
+                  {
+                    writeConcern: {
+                      w: !isNaN(sthConfig.WRITE_CONCERN) ?
+                        parseInt(sthConfig.WRITE_CONCERN, 10) : sthConfig.WRITE_CONCERN
+                    }
+                  },
+                  callback
+                );
+              }
+            );
+          } else {
+            process.nextTick(callback);
+          }
+        }
+      );
+    }
+  );
+}
+
+/**
+ * Removes the data associated to the attributes specified in the provided data in case of collection per
+ *  service path data model
+ * @param {object} data It is an object including the following properties:
+ *  - {string} service The service
+ *  - {string} servicePath The service path
+ *  - {string} entityId The entity id
+ *  - {string} entityType The entity type
+ *  - {string} attrName The attribute name. In case no attribute name is provided, all the attributes of the
+ *    provided entity are removed
+ * @param {object} options It is an object including the following properties:
+ *  - {boolean} isAggregated Flag indicating if the operation refers to aggregated data. Raw data otherwise
+ * @param callback The callback to call with error or the result of the operation
+ */
+function removeDataIfCollection4ServicePath(data, options, callback) {
+  var service = data.service,
+      servicePath = data.servicePath,
+      entityId = data.entityId,
+      entityType = data.entityType,
+      attrName = data.attrName;
+  var isAggregated = options.isAggregated;
+
+  if (entityId || attrName) {
+    getCollection(
+      {
+        service: service,
+        servicePath: servicePath
+      },
+      {
+        isAggregated: isAggregated,
+        shouldCreate: false,
+        shouldStoreHash: false,
+        shouldTruncate: false
+      },
+      function (err, collection) {
+        if (err) {
+          if (callback) {
+            return process.nextTick(callback.bind(null, err));
+          }
+        } else {
+          var findCondition;
+          if (isAggregated) {
+            findCondition = {
+              '_id.entityId': entityId,
+              '_id.entityType': entityType
+            };
+            if (attrName) {
+              findCondition['_id.attrName'] = attrName;
+            }
+          } else {
+            findCondition = {
+              entityId: entityId,
+              entityType: entityType
+            };
+            if (attrName) {
+              findCondition.attrName = attrName;
+            }
+          }
+          collection.remove(
+            findCondition,
+            {
+              writeConcern: {
+                w: !isNaN(sthConfig.WRITE_CONCERN) ? parseInt(sthConfig.WRITE_CONCERN, 10) : sthConfig.WRITE_CONCERN
+              }
+            },
+            callback
+          );
+        }
+      }
+    );
+  } else {
+    var databaseName = getDatabase(service);
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-sth-comet/commit/787c2bfa1902e97d6934f87c6443edc0b05a17be